src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -473,6 +481,11 @@ bool CDSConfig::check_vm_args_consistency(bool patch_mod_javabase, bool mode_fla
   return true;
 }
 
+bool CDSConfig::allow_only_single_java_thread() {
+  // See comments in JVM_StartThread()
+  return is_dumping_static_archive();
```

Comment:
The test in `JVM_StartThread()` calls `CDSConfig::is_dumping_static_archive()`. Should it be updated to call `CDSConfig::allow_only_single_java_thread()`?

Reply:
I updated `JVM_StartThread()`  to call `CDSConfig::allow_only_single_java_thread()`